src/SIM/ModelGenerator.C

```diff
@@ -21,6 +21,32 @@
 #include <array>
 
 
+/*!
+  \brief Static helper adding knot vectors to a g2 definition.
+  \param g2 The string to add knot vectors to
+  \param[in] num The number of knot vectors to add
+  \param[in] geo XML element describing geometry
+*/
+
+static void AddKnots(std::string& g2, size_t num, const TiXmlElement* geo)
+{
+  std::string min = "umin", max = "umax";
+  for (size_t i = 0; i < num; ++i, ++min[0], ++max[0]) {
+    double pmin = 0.0, pmax = 1.0;
+    bool print = false;
+    print |= utl::getAttribute(geo,min.c_str(),pmin);
+    print |= utl::getAttribute(geo,max.c_str(),pmax);
+    if (print)
+      IFEM::cout <<"\n\t" << min[0] << "["<< pmin << ","<< pmax <<"]";
+    std::stringstream str;
+    str << pmin << " " << pmin << " " << pmax << " " << pmax;
+    g2.append("\n2 2\n");
+    g2.append(str.str());
+  }
+  g2.append("\n");
```

Comment:
Alternative rewrite (for compactness, but not tested)
```
  char* min = const_cast<char*>("umin");
  char* max = const_cast<char*>("umax");
  std::stringstream str;
  for (size_t i = 0; i < num; ++i, ++min[0], ++max[0]) {
    double pmin = 0.0, pmax = 1.0;
    if (utl::getAttribute(geo,min,pmin) | utl::getAttribute(geo,max,pmax))
      IFEM::cout <<"\n\t"<< min[0] <<"["<< pmin <<","<< pmax <<"]";
    str <<"\n2 2\n"<< pmin <<" "<< pmin <<" "<< pmax <<" "<< pmax;
  }
  g2.append(str.str());
  g2.append("\n");
```

Reply:
took most of it, but i cannot do those char* shenanigans. cdata is in read only memory, it instantly segfaults.